crates/new_task/src/dependency_task.rs

```diff
@@ -0,0 +1,140 @@
+#[cfg(async_executor_impl = "async-std")]
+use async_std::task::{spawn, JoinHandle};
+#[cfg(async_executor_impl = "tokio")]
+use tokio::task::{spawn, JoinHandle};
+
+use futures::Future;
+
+use crate::dependency::Dependency;
+
+/// Defines a type that can handle the result of a dependency
+pub trait HandleDepResult: Send + Sized + Sync + 'static {
```

Comment:
Nit: Can we switch this to something like `HandleDepOutput` and `HandleDepOutput::handle_output` as well? Sorry for not wording this more explicitly in my original comment.
My reasons for requesting this is that reading "result" in Rust code makes me - and I assume most other Rust users - think that this abstraction deals explicitly with fallible computation, which isn't the case. I don't much care about exact names here, my main concern is reserving "result" for something that explicitly deals with errors

Reply:
Yeah this makes sense, I had forgotten that I named this "result"